Dockerfile

```diff
@@ -1,21 +1,27 @@
-FROM golang:1.13 as build
-
-ENV GO111MODULE=on
-
-WORKDIR /app
-
-COPY go.mod .
-COPY go.sum .
-
+# Build the manager binary
+FROM golang:1.13 as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
 RUN go mod download
 
-COPY . .
-
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o authorino main.go
+# Copy the go source
+COPY main.go main.go
+COPY api/ api/
+COPY controllers/ controllers/
 
-FROM gcr.io/distroless/base
-COPY --from=build /app/authorino /
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go
 
-EXPOSE 50051
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:nonroot
+WORKDIR /
+COPY --from=builder /workspace/manager .
```

Comment:
Why is it called "manager" instead of the name of the app? Is this a convention?

Reply:
Same as the previous comment :)